### PR TITLE
Category

### PR DESCRIPTION
**Pull request by OlegKandalov**

### CODE REVIEW

File: src/main/java/com/exadel/discount/platform/model/dto/SubCategoryDto.java [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
 import javax.validation.constraints.Size;
 import java.util.List;
+import java.util.UUID;
 
 @Data
 public class SubCategoryDto {
     @NotBlank
     @Size(min = 3, max = 20)
     private String name;
+    private UUID categoryId;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/model/dto/SubCategoryDto.java:16]: `@NotNull`

File: src/main/java/com/exadel/discount/platform/model/dto/VendorDto.java [hunk 2]
```diff
@@ -21,9 +19,4 @@
     private String email;
     @NotBlank
     private String image;
-    @ToString.Exclude
-    private List<DiscountDto> discountDtos;
-    @ToString.Exclude
-    private List<VendorLocationDto> locationDtos;
-    private boolean deleted;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/model/dto/VendorDto.java:28]: leave `deleted`

File: src/main/java/com/exadel/discount/platform/model/dto/VendorLocationDto.java [hunk 3]
```diff
@@ -17,8 +18,7 @@
     private String city;
     @Size(min = 2, max = 50)
     private String addressLine;
-    private VendorDto vendorDto;
-    private boolean deleted;
+    private UUID vendorId;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/model/dto/VendorLocationDto.java:21]: `NotNull`

File: src/main/resources/liquibase/29.06.2021_oleg_category_relationship_with_subCategory.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                <constraints nullable="false"/>
+            </column>
+            <column name="c_deleted" type="boolean"/>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="add_column_to_sub_category" author="oleg">
+        <addColumn tableName="sub_category">
+            <column name="category_id" type="uuid">
+                <constraints nullable="false"/>
```
Comment (ohalushchak-exadel) [inline src/main/resources/liquibase/29.06.2021_oleg_category_relationship_with_subCategory.xml:21]: this won't work if there are any sub categories in the database
either delete all existing sub categories or create a `default` category and use it as a default value in the existing sub categories

File: src/main/java/com/exadel/discount/platform/model/dto/VendorDto.java (hunk 2 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/model/dto/VendorDto.java:28]: i have this field in the vendorResponse file

**Review by ohalushchak-exadel — APPROVED** [review]

**Review by nkharitoniuk — APPROVED** [review]

**Pull request merged**